cardano-cli/test/cardano-cli-test/Test/Cli/CreateTestnetData.hs

```diff
@@ -19,11 +34,60 @@ hprop_create_testnet_data_minimal =
 
     let outputDir = tempDir </> "out"
 
+    -- We test that the command doesn't crash, because otherwise
+    -- execCardanoCLI would fail.
     H.noteM_ $ execCardanoCLI
       ["conway",  "genesis", "create-testnet-data"
       , "--testnet-magic", "42"
       , "--out-dir", outputDir
       ]
+    success
 
-    -- We test that the command doesn't crash, because otherwise
-    -- execCardanoCLI would fail.
+hprop_create_testnet_data_create_nonegative_supply :: Property
+hprop_create_testnet_data_create_nonegative_supply = do
+  -- FIXME rewrite this as a property test
+  let supplyValues =
+        [ -- (total supply, delegated supply)
+          (2_000_000_000, 1_000_000_000)
+        , (1_000_000_000, 1_000_000_000)
+        , (1_000_000_000, 2_000_000_000)
+        ] :: [(Int, Int)]
+
+  propertyOnce $ forM_ supplyValues $ \(totalSupply, delegatedSupply) ->
+    moduleWorkspace "tmp" $ \tempDir -> do
+      let outputDir = tempDir </> "out"
+
+      H.noteM_ $ execCardanoCLI
+        ["conway",  "genesis", "create-testnet-data"
+        , "--testnet-magic", "42"
+        , "--pools", "3"
+        , "--total-supply", show totalSupply
+        , "--delegated-supply", show delegatedSupply
+        , "--stake-delegators", "3"
+        , "--utxo-keys", "3"
+        , "--drep-keys", "3"
+        , "--out-dir", outputDir
+        ]
+
+      testGenesis@TestGenesis{maxLovelaceSupply, initialFunds} <- H.leftFailM . readJsonFile $ outputDir </> "genesis.json"
+      H.note_ $ show testGenesis
+
+      H.note_ "check that max lovelace supply is positive"
+      H.assertWith maxLovelaceSupply (>= 0)
```

Comment:
Can't you check that `maxLovelaceSupply` is the total supply?

Reply:
done